src/mlpack/methods/ann/layer/lstm.hpp

```diff
@@ -147,52 +117,44 @@ class LSTM
    * @param error The calculated error.
    * @param gradient The calculated gradient.
    */
-  template<typename InputType, typename ErrorType, typename GradientType>
-  void Gradient(const InputType& input,
-                const ErrorType& error,
-                GradientType& gradient);
+  void Gradient(const MatType& input,
+                const MatType& error,
+                MatType& gradient);
 
-  //! Get the maximum number of steps to backpropagate through time (BPTT).
-  size_t Rho() const { return rho; }
-  //! Modify the maximum number of steps to backpropagate through time (BPTT).
-  size_t& Rho() { return rho; }
+  /*
+   * Reset the layer parameter.
+   */
+  void Reset();
+
+  /**
+   * Reset the recurrent state of the LSTM layer, and allocate enough space to
+   * hold `bpttSteps` of previous passes with a batch size of `batchSize`.
+   *
+   * @param bpttSteps Number of steps of history to allocate space for.
+   * @param batchSize Batch size to prepare for.
+   */
+  void ClearRecurrentState(const size_t bpttSteps, const size_t batchSize);
 
   //! Get the parameters.
-  OutputDataType const& Parameters() const { return weights; }
+  const MatType& Parameters() const { return weights; }
   //! Modify the parameters.
-  OutputDataType& Parameters() { return weights; }
-
-  //! Get the output parameter.
-  OutputDataType const& OutputParameter() const { return outputParameter; }
-  //! Modify the output parameter.
-  OutputDataType& OutputParameter() { return outputParameter; }
+  MatType& Parameters() { return weights; }
 
-  //! Get the delta.
-  OutputDataType const& Delta() const { return delta; }
-  //! Modify the delta.
-  OutputDataType& Delta() { return delta; }
-
-  //! Get the gradient.
-  OutputDataType const& Gradient() const { return grad; }
-  //! Modify the gradient.
-  OutputDataType& Gradient() { return grad; }
-
-  //! Get the number of input units.
-  size_t InSize() const { return inSize; }
-
-  //! Get the number of output units.
-  size_t OutSize() const { return outSize; }
-
-  //! Get the size of the weights.
   size_t WeightSize() const
   {
+    // TODO ...
     return (4 * outSize * inSize + 7 * outSize + 4 * outSize * outSize);
   }
 
-  //! Get the shape of the input.
-  size_t InputShape() const
+  void ComputeOutputDimensions()
   {
-    return inSize;
+    inSize = std::accumulate(this->inputDimensions.begin(),
+        this->inputDimensions.end(), 0);
```

Comment:
Ig this has missing include statement of \<numeric\>

Reply:
Thanks!  I added it to `mlpack/prereqs.hpp` in 6eecba768. :+1: